src/Core/Batch/BatchDaemon.php

```diff
@@ -0,0 +1,203 @@
+<?php
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Core\Batch;
+
+/**
+ * An external daemon script for executing the batch jobs.
+ *
+ * @codeCoverageIgnore
+ *
+ * The system test is responsible for testing this class.
+ * {@see \Google\Cloud\Tests\System\Core\Batch}
+ */
+class BatchDaemon
+{
+    use SysvTrait;
+    use HandleFailureTrait;
+
+    /* @var BatchRunner */
+    private $runner;
+    
+    /* @var boolean */
+    private $shutdown;
+
+    /* @var array */
+    private $descriptorSpec;
+
+    /* @var string */
+    private $command;
+
+    /**
+     * Prepare the descriptor spec and install signal handlers.
+     *
+     * @param string $entrypoint Daemon's entrypoint script.
+     */
+    public function __construct($entrypoint)
+    {
+        if (! $this->isSysvIPCLoaded()) {
+            throw new \RuntimeException('SystemV IPC exntensions are missing.');
+        }
+        $this->runner = new BatchRunner(
+            new SysvConfigStorage(),
+            new SysvSubmitter()
+        );
+        $this->shutdown = false;
+        // Just share the usual descriptors.
+        $this->descriptorSpec = [
+            0 => ['file', 'php://stdin', 'r'],
+            1 => ['file', 'php://stdout', 'w'],
+            2 => ['file', 'php://stderr', 'w']
+        ];
+        // setup signal handlers
+        pcntl_signal(SIGTERM, [$this, "sigHandler"]);
+        pcntl_signal(SIGINT, [$this, "sigHandler"]);
+        pcntl_signal(SIGHUP, [$this, "sigHandler"]);
+        pcntl_signal(SIGALRM, [$this, "sigHandler"]);
+        $this->command = sprintf('exec php -d auto_prepend_file="" %s', $entrypoint);
```

Comment:
The other PR contains a file for setting as `auto_prepend_file`, which basically catch all the errors and report it as a Log entry. We need to prevent the infinite loop of this.